stdlib/public/core/Arrays.swift.gyb

```diff
@@ -1210,6 +1210,54 @@ extension ${Self} : RangeReplaceableCollection, _ArrayProtocol {
   /// greater than the requested capacity. Use the array's `capacity` property
   /// to determine the size of the new storage.
   ///
+  /// Preserving an Array's Geometric Growth Strategy
+  /// ===============================================
+  ///
+  /// If you implement an custom data structure backed by an array that grows
+  /// dynamically, naively calling the `reserveCapacity(_:)` method can lead
+  /// to lower than expected performance. Arrays need to follow a geometric
+  /// allocation pattern for appending elements to have amortized
+  /// constant-time performance. The `Array` type's `append(_:)` and
+  /// `append(contentsOf:)` methods take care of this detail for you, but
+  /// `reserveCapacity(_:)` allocates only as much space as you tell it to
+  /// (padded to a round value), and no more. This avoids over-allocation, but
+  /// can result in insertion not having amortized constant-time performance.
+  ///
+  /// The following code declares `values`, an array of integers, and the
+  /// `addTenQuadratic()` function, which adds ten more values to the `values`
+  /// array on each call.
+  ///
+  ///       var values: [Int] = [0, 1, 2, 3]
+  ///
```

Comment:
Do we have a standard method of marking code snippets that are showing "what not to do"? It would be nice to be explicit for someone skimming the text with a `// don't do this`, etc.